src/azul/openapi.py

```diff
@@ -37,32 +109,52 @@ def annotated_specs(gateway_id, app, toplevel_spec) -> JSON:
     specs = aws.api_gateway_export(gateway_id)
     clean_specs(specs)
     specs = merge_dicts(toplevel_spec, specs, override=True)
-    docs = get_doc_specs(app)
-    for path, doc_spec in docs.items():
-        for verb, description in specs['paths'][path].items():
-            description = merge_dicts(description, doc_spec)
-            specs['paths'][path][verb] = description
-    return specs
+    path_specs, method_specs = get_app_specs(app)
+    return join_specs(specs, path_specs, method_specs)
 
 
 def clean_specs(specs):
     """
     Adjust specs from API Gateway so they pass linting
     """
     # Filter out 'options' since it causes linting errors
-    for path in specs['paths']:
-        specs['paths'][path].pop('options', None)
+    for path in specs['paths'].values():
+        path.pop('options', None)
     # Remove listed servers since API Gateway give false results
     specs.pop('servers')
 
 
-def get_doc_specs(app):
-    docs = {}
-    for route, entries in app.routes.items():
+def get_app_specs(app):
+    path_specs = {}
+    method_specs = {}
+    for entries in app.routes.values():
         func = next(iter(entries.values())).view_function
-        if hasattr(func, 'api_spec'):
-            docs[route] = func.api_spec
-    return docs
+        if hasattr(func, 'path_specs'):
+            for path, spec in func.path_specs.items():
+                if path in path_specs:
```

Comment:
I noticed you do this here and there. It's not explicitly specified in the contribution guide but we follow the commonly accepted [EAFP](https://stackoverflow.com/questions/11360858/what-is-the-eafp-principle-in-python) idiom. If you feel it should be mentioned, could you add a section to the guidelines? Either way, though, this should use EAFP.

Reply:
Sure. I find LBYL to be more legible (and often less code), but I understand the strengths of EAFP, especially with regard to race conditions. I'm happy to switch it over.

Reply:
Despite EAFP being more verbose, it IS easier to reason about because it doesn't do everything twice.